File: 317/c.cpp
#include <iostream>
#include <vector>
#include <string>
#include <stack>
#include <map>
#include <queue>
using namespace std;

#define rep(i, n) for(int i=0; i<n; i++)
#define ll long long

struct Edge {
    int to, cost, dist;
    Edge(int to=0, int cost=0): to(to), cost(cost) {}
};

int ans;
vector<bool> visited(11, false);

void dfs(vector<vector<Edge> > &g, int &pos, int dist) {
    visited[pos] = true;
    ans = max(ans, dist);
    for(auto v : g[pos]) {
        if(!visited[v.to]) {
            dfs(g, v.to, dist+v.cost);
        }
    }
    visited[pos] = false;
}

int main() {
    int n, m;
    cin >> n >> m;
    int a, b, c;
    vector<vector<Edge> > g(n);

    rep(i, m) {
        cin >> a >> b >> c;
        a--;
        b--;
        g[a].emplace_back(b, c);
        g[b].emplace_back(a, c);
    }

    rep(i, n) dfs(g, i, 0);
    cout << ans << endl;
    return 0;
}

File: 317/C.py
# dfs(深さ優先探索)を使用して実装
# CPythonでは通らなかったがPyPyでは通った
# この二つの違い何？？
N, M = map(int, input().split())

G = [[0] * (N+1) for _ in range(N+1)]

for _ in range(M):
    a, b , c = map(int, input().split())
    G[a][b] = c
    G[b][a] = c

ans = 0
visited = [False] * (N+1)

def dfs(pos, dist):
    global ans
    visited[pos] = True
    ans = max(ans, dist)
    for i in range(1, N+1):
        if not visited[i] and G[pos][i] != 0:
            dfs(i, dist+G[pos][i])
    visited[pos] = False

for i in range(1, N+1):
    dfs(i, 0)

print(ans)